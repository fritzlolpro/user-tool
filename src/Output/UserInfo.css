.user-pic {
  margin-bottom: 16px !important;
}
.user-name {
  font-family: Roboto, sans-serif !important;
  font-weight: 400 !important;
  margin-bottom: 8px !important;
  margin-top: 8px !important;
  font-size: 32px !important;
  color: #616161 !important;
  text-transform: capitalize !important;
}
.user-plate {
  height: 100% !important;
  position: relative !important;
}
.actions-block {
  position: absolute !important;
  bottom: 0 !important;
  left: 0 !important;
  right: 0 !important;
  margin: auto !important;
}
