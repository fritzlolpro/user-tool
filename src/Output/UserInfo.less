.user-plate() {
    height: 100%;
    position: relative;
}

.actions-block() {
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    margin: auto;
}

.user-pic() {
    margin-bottom: 16px;
}

.user-name() {
    font-family: Roboto, sans-serif;
    font-weight: 400;
    margin-bottom: 8px;
    margin-top: 8px;
    font-size: 32px;
    color: #616161;

    text-transform: capitalize
}

.user-pic {
    .user-pic() !important;
}

.user-name {
    .user-name() !important;
}

.user-plate {
    .user-plate() !important;
}

.actions-block {
    .actions-block() !important;
}