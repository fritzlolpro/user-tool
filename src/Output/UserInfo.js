import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';
import Avatar from 'material-ui/Avatar';
import './UserInfo.css';

const UserInfo = ({user, onEdit, onDelete}) => {
    const onHandleEdit = () => onEdit(user.id)
    const onHandleDelete = () => onDelete(user.id)
    return (
        <div className='user-plate'>
            <Avatar className='user-pic' src={user.picture} size={180}/>
            <div>
                <p className='user-name'>{user.name.first}</p>
                <p className='user-name'>{user.name.last}</p>
            </div>

            <div className='actions-block'>
                <RaisedButton label="Edit" onClick={onHandleEdit} primary={true}/>
                <FlatButton label="Delete" onClick={onHandleDelete}/>
            </div>
        </div>
    )
}

export default UserInfo