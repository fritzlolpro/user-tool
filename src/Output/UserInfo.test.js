import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import UserInfo from './UserInfo';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

describe('UserInfo', () => {
    const props = {
        user: {
            id: 123,
            name: {
                first: 'TestFirst',
                last: 'TestLast'
            },
            picture: 'https://randomuser.me/api/portraits/men/83.jpg'
        }
    }
    it('should be defined', () => {
        expect(UserInfo).toBeDefined();
    });

    it('renders without crashing'),
    () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <MuiThemeProvider>
            <UserInfo {...props}/>
        </MuiThemeProvider>, div);
    }

    test('has a valid snapshot', () => {
        const component = renderer.create(
            <MuiThemeProvider>
                <UserInfo {...props}/>
            </MuiThemeProvider>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})
