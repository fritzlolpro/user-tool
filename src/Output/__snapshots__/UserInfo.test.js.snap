// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo has a valid snapshot 1`] = `
<div
  className="user-plate"
>
  <img
    className="user-pic"
    size={180}
    src="https://randomuser.me/api/portraits/men/83.jpg"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(188, 188, 188)",
        "borderRadius": "50%",
        "color": "#ffffff",
        "display": "inline-flex",
        "fontSize": 90,
        "height": 180,
        "justifyContent": "center",
        "muiPrepared": true,
        "userSelect": "none",
        "width": 180,
      }
    }
  />
  <div>
    <p
      className="user-name"
    >
      TestFirst
    </p>
    <p
      className="user-name"
    >
      TestLast
    </p>
  </div>
  <div
    className="actions-block"
  >
    <div
      className={undefined}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": 2,
          "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                 0 1px 4px rgba(0, 0, 0, 0.12)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "minWidth": 88,
          "muiPrepared": true,
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        }
      }
    >
      <button
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "#00bcd4",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": "100%",
          }
        }
        tabIndex={0}
        type="button"
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "backgroundColor": false,
                "borderRadius": 2,
                "height": 36,
                "muiPrepared": true,
                "top": 0,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              }
            }
          >
            <span
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 14,
                  "fontWeight": 500,
                  "letterSpacing": 0,
                  "margin": 0,
                  "muiPrepared": true,
                  "opacity": 1,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "position": "relative",
                  "textTransform": "uppercase",
                  "userSelect": "none",
                }
              }
            >
              Edit
            </span>
          </div>
        </div>
      </button>
    </div>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": 88,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          Delete
        </span>
      </div>
    </button>
  </div>
</div>
`;
